### PR TITLE
pythonPackages.aws-sam-translator: init at 1.5.4

### PR DESCRIPTION
**Pull request by andreabedini**
###### Motivation for this change

New python package, needed for `aws-sam-cli` which I am about to submit as well.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+buildPythonPackage rec {
+  pname = "aws-sam-translator";
+  version = "1.5.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "9d8a25e058c78d2cef5c07aec7f98cbc2070dbfc2eb6a2e102a16beafd14e3ca";
+  };
+
+  # Tests are not included in the PyPI package
```
Comment (dotlambda) [inline pkgs/development/python-modules/aws-sam-translator/default.nix]: You may want to use `fetchFromGitHub` in that case.

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    boto3
+    enum34
+    jsonschema
+    six
+  ];
+
+  meta = {
+    homepage = https://github.com/awslabs/aws-sam-cli;
```
Comment (dotlambda) [inline pkgs/development/python-modules/aws-sam-translator/default.nix]: https://github.com/awslabs/serverless-application-model

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please create a separate commit that adds you to `maintainer-list.nix` and name the second commit exactly as this PR.

File: pkgs/development/python-modules/aws-sam-translator/default.nix (hunk 1 above)
Comment (andreabedini) [reply pkgs/development/python-modules/aws-sam-translator/default.nix]: I looked into it but the versions don't line up, latest tag on GitHub is 1.5 but latest version on PyPi is 1.5.4. The released version is probably just master but they really should tag it.

File: pkgs/development/python-modules/aws-sam-translator/default.nix (hunk 2 above)
Comment (andreabedini) [reply pkgs/development/python-modules/aws-sam-translator/default.nix]: upsie! fixed

**Comment by andreabedini** [comment]
Rebased to split d3953f8 into two separate commits.

File: pkgs/development/python-modules/aws-sam-translator/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/aws-sam-translator/default.nix]: Well, then better stick with `fetchPypi`.

**Comment by dotlambda** [comment]
The commit messages still need to be changed.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.aws-sam-translator python3.pkgs.aws-sam-translator

**Comment by dotlambda** [comment]
> Could not find a version that satisfies the requirement enum34~=1.1

`enum34` is probably a `python3`-only dependency

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    inherit pname version;
+    sha256 = "9d8a25e058c78d2cef5c07aec7f98cbc2070dbfc2eb6a2e102a16beafd14e3ca";
+  };
+
+  # Tests are not included in the PyPI package
+  doCheck = false;
+
+  propagatedBuildInputs = [
+    boto3
+    enum34
```
Comment (FRidh) [inline pkgs/development/python-modules/aws-sam-translator/default.nix]: enum34 should be added conditionally, that is, only with Python 2.

**Review by FRidh — CHANGES_REQUESTED** [review]
@dotlambda 
> Python 3.4 Enum backported to 3.3, 3.2, 3.1, 2.7, 2.6, 2.5, and 2.4

File: pkgs/development/python-modules/aws-sam-translator/default.nix (hunk 3 above)
Comment (andreabedini) [reply pkgs/development/python-modules/aws-sam-translator/default.nix]: I don't think this package supports Python3 at all, I need to express it somewhere

**Comment by andreabedini** [comment]
> The commit messages still need to be changed.

sorry @dotlambda I'm not sure how they should be changed, is it documented in the docs? I'll check again

File: pkgs/development/python-modules/aws-sam-translator/default.nix (hunk 3 above)
Comment (Assassinkin) [reply pkgs/development/python-modules/aws-sam-translator/default.nix]: @andreabedini  use `disabled = isPy3k;`

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.aws-sam-translator python3.pkgs.aws-sam-translator

**Pull request merged**